http-clients/src/test/java/com/palantir/remoting3/clients/ClientConfigurationsTest.java

```diff
@@ -67,4 +71,39 @@ public void testFromParameters_fillsInDefaults() {
         assertThat(actual.enableGcmCipherSuites()).isFalse();
         assertThat(actual.proxy().select(URI.create("https://foo"))).containsExactly(Proxy.NO_PROXY);
     }
+
+    @Test
+    public void meshProxy_maxRetriesMustBe0() throws Exception {
+        ServiceConfiguration invalidServiceConfig = ServiceConfiguration.builder()
+                .uris(uris)
+                .security(SslConfiguration.of(Paths.get("src/test/resources/trustStore.jks")))
+                .proxy(ProxyConfiguration.mesh("localhost:1234"))
+                .maxNumRetries(2)
+                .build();
+        assertThatThrownBy(() -> ClientConfigurations.of(invalidServiceConfig))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("If meshProxy is configured then maxNumRetries must be 0");
+
+        ClientConfiguration validConfig = ClientConfigurations.of(ServiceConfiguration.builder()
+                .uris(uris)
+                .security(SslConfiguration.of(Paths.get("src/test/resources/trustStore.jks")))
+                .proxy(ProxyConfiguration.mesh("localhost:1234"))
+                .maxNumRetries(0)
+                .build());
+        assertThat(validConfig.meshProxy()).isEqualTo(Optional.of(HostAndPort.fromParts("localhost", 1234)));
+        assertThat(validConfig.maxNumRetries()).isEqualTo(0);
+    }
+
+    @Test
+    public void meshProxy_exactlyOneUri() throws Exception {
+        ServiceConfiguration invalidServiceConfig = ServiceConfiguration.builder()
```

Comment:
I'm fine with making a test-local factory method here to keep the tests shorter. (didn't want to bloat the public API main/ with another factory)

Reply:
done